web/stream/service.go

```diff
@@ -12,47 +12,52 @@ import (
 // To add a new service, add a new field to this struct and
 // initialize the service in the NewStreamServices function.
 type StreamServices struct {
-	Submission *Service[qf.Submission]
+	Submission *Service[qf.Submission, uint64]
 }
 
 // NewStreamServices creates a new StreamServices.
 func NewStreamServices() *StreamServices {
 	return &StreamServices{
-		Submission: NewService[qf.Submission](),
+		Submission: NewService[qf.Submission, uint64](),
 	}
 }
 
+// ID is the allowed type for stream IDs.
+type ID interface {
+	uint64 | string
+}
+
 // Service[T] is a type specific stream service.
 // It also contains a map of streams that are currently connected.
-type Service[T any] struct {
+type Service[T any, V ID] struct {
```

Comment:
I guess a common pattern for map-based structs is to have the key `K` first and then the value `V`:
```go
type Service[K ID, V any] struct {
```